# Capgemini Engineering. -- AU: 2022-2023

# Ecole Supérieure de la Statistique et Analyse de l'Information

**Stage Ingénieur**

**Projet 2-- Deep learning in the automotive industry**

**Realisé par: Yosser Akili**

**Supervisé par: Ines Haddad Saidani**

# 📖 Introduction



Les voitures ont longtemps été un pilier incontournable de la société moderne, révolutionnant la façon dont nous nous déplaçons, interagissons et percevons le monde qui nous entoure. Depuis leur création, les automobiles ont évolué bien au-delà de simples moyens de transport pour devenir des symboles de style, d'innovation technologique et de mode de vie. Elles ont créé des rêves d'aventures sur la route et sont devenues des éléments clés de l'identité culturelle et sociale.

Les constructeurs automobiles du monde entier rivalisent pour créer des véhicules qui incarnent à la fois la performance, le confort et le design. De ce fait, une multitude de modèles de voitures ont vu le jour pour répondre aux besoins variés des consommateurs, reflétant les préférences esthétiques, les styles de vie et les besoins fonctionnels.

Dans ce fichier, je vais traiter des différentes caractéristiques de voitures pour enfin créer une base et pouvoir prédire à partir de cette base si une voiture est une bonne affaire ou mauvaise affaire.

# I.Imporation des libraries

In [1]:
import os
import numpy as np
import pandas as pd
import seaborn as sns
import re
import matplotlib.pyplot as plt
!pip install missingno
import missingno
import warnings

# II. Compréhension des données
**Dans cette partie on va voir quatre sous-parties qui sont :**

* **1. Présentation de la base.**


* **2. Statistiques descriptives**


* **3. Nettoyage de la base**


* **4. vérification de la qualité des données.**

# 1. Présentation de la base

Les données utilisées sont extraites à partir du site kbb.com (kelley blue book). En fait, **kbb.com** est un site web dédié à l'achat et à la vente de voitures neuves et d'occasion, ainsi qu'à la location de voitures. Kelley Blue Book est une société d'évaluation de véhicules et de recherche automobile basée à Irvine, en Californie, reconnue à la fois par les consommateurs et l'industrie automobile. La société appartient à la filiale Cox Automotive de Cox Enterprises.

Sur kbb.com, les acheteurs peuvent rechercher des voitures en fonction de l'année de fabrication , la marque, le modèle et d'autres critères. Les vendeurs peuvent également publier des annonces pour vendre leurs voitures sur le site.

En plus de la recherche et de l'achat, Cars.com propose également des services d'évaluation de voitures.



In [2]:
acura_cars= pd.read_excel('acura_cars.xlsx')
alfa_romeo_cars= pd.read_excel('alfa_romeo_cars.xlsx')
audi_cars= pd.read_excel('audi_cars.xlsx')
bmw_cars= pd.read_excel('bmw_cars.xlsx')
buick_cars= pd.read_excel('buick_cars.xlsx')
cadillac_cars= pd.read_excel('cadillac_cars.xlsx')
chevrolet_cars= pd.read_excel('chevrolet_cars.xlsx')
chrysler_cars= pd.read_excel('chrysler_cars.xlsx')
dodge_cars= pd.read_excel('dodge_cars.xlsx')
eagle_cars= pd.read_excel('eagle_cars.xlsx')
fiat_cars= pd.read_excel('fiat_cars.xlsx')
fisker_cars= pd.read_excel('fisker_cars.xlsx')
ford_cars= pd.read_excel('ford_cars.xlsx')
freightliner_cars= pd.read_excel('freightliner_cars.xlsx')
genesis_cars= pd.read_excel('genesis_cars.xlsx')
gmc_cars= pd.read_excel('gmc_cars.xlsx')
honda_cars= pd.read_excel('honda_cars.xlsx') 
hummer_cars= pd.read_excel('hummer_cars.xlsx') 
hyundai_cars= pd.read_excel('hyundai_cars.xlsx')
infiniti_cars= pd.read_excel('infiniti_cars.xlsx')
isuzu_cars= pd.read_excel('isuzu_cars.xlsx')
jeep_cars= pd.read_excel('jeep_cars.xlsx')
kia_cars= pd.read_excel('kia_cars.xlsx')
land_rover_cars= pd.read_excel('land_rover_cars.xlsx')
lincoln_cars= pd.read_excel('lincoln_cars.xlsx')
lotus_cars= pd.read_excel('lotus_cars.xlsx')
lucid_cars= pd.read_excel('lucid_cars.xlsx')
mazda_cars= pd.read_excel('mazda_cars.xlsx')
mercedes_benz_cars= pd.read_excel('mercedes_benz_cars.xlsx')
mercury_cars= pd.read_excel('mercury_cars.xlsx')
mini_cars= pd.read_excel('mini_cars.xlsx')
mitsubishi_cars= pd.read_excel('mitsubishi_cars.xlsx')
nissan_cars= pd.read_excel('nissan_cars.xlsx')
oldsmobile_cars= pd.read_excel('oldsmobile_cars.xlsx')
panoz_cars= pd.read_excel('panoz_cars.xlsx')
plymouth_cars= pd.read_excel('plymouth_cars.xlsx')
polestar_cars= pd.read_excel('polestar_cars.xlsx')
pontiac_cars= pd.read_excel('pontiac_cars.xlsx')
saab_cars= pd.read_excel('saab_cars.xlsx')
smart_cars= pd.read_excel('smart_cars.xlsx')
srt_cars= pd.read_excel('srt_cars.xlsx')
subaru_cars= pd.read_excel('subaru_cars.xlsx')
suzuki_cars= pd.read_excel('suzuki_cars.xlsx')
toyota_cars= pd.read_excel('toyota_cars.xlsx')
volkswagen_cars= pd.read_excel('volkswagen_cars.xlsx')

### a. Fusionner plusieurs fichiers Excel en un seul DataFrame pandas

In [3]:
# merge files
data = pd.concat([acura_cars,alfa_romeo_cars,audi_cars,bmw_cars,buick_cars, cadillac_cars, chevrolet_cars, chrysler_cars, dodge_cars,eagle_cars,fiat_cars,fisker_cars, ford_cars,freightliner_cars,genesis_cars,gmc_cars,honda_cars ,hummer_cars ,hyundai_cars,infiniti_cars,isuzu_cars,jeep_cars,kia_cars,land_rover_cars,lincoln_cars,lotus_cars,lucid_cars,mazda_cars,mercedes_benz_cars,mercury_cars,mini_cars,mitsubishi_cars,nissan_cars,oldsmobile_cars,panoz_cars,plymouth_cars,polestar_cars,pontiac_cars,saab_cars,smart_cars,srt_cars,subaru_cars,suzuki_cars,toyota_cars,volkswagen_cars ], axis=0)
print(data)

     Unnamed: 0                      Car    Price Value Performance Quality  \
0             1           2024 Acura MDX  $51,045   4.3         4.3     4.4   
1             2       2024 Acura Integra  $32,695   5.0         5.0     5.0   
2             3           2023 Acura RDX  $42,945   3.9         4.3     4.0   
3             4           2023 Acura MDX  $50,745   4.3         4.3     4.4   
4             5           2023 Acura TLX  $41,245   4.5         4.5     4.7   
..          ...                      ...      ...   ...         ...     ...   
273         274     1992 Volkswagen Golf   $2,165   4.6         4.6     4.6   
274         275   1992 Volkswagen Passat   $2,172   4.4         4.1     4.2   
275         276      1992 Volkswagen Fox   $2,530   4.6         4.4     4.4   
276         277      1992 Volkswagen GTI   $2,789   4.7         4.7     4.6   
277         278  1992 Volkswagen Corrado   $4,507   4.8         4.9     4.9   

     Comfort  Reliability  Styling  \
0        4.6 

### b. Voir les cinq premières lignes et les cinq dernières :

In [4]:
data.head()

Unnamed: 0                 Car    Price Value Performance Quality  Comfort  \
0           1      2024 Acura MDX  $51,045   4.3         4.3     4.4      4.6   
1           2  2024 Acura Integra  $32,695   5.0         5.0     5.0      5.0   
2           3      2023 Acura RDX  $42,945   3.9         4.3     4.0      4.3   
3           4      2023 Acura MDX  $50,745   4.3         4.3     4.4      4.6   
4           5      2023 Acura TLX  $41,245   4.5         4.5     4.7      4.4   

   Reliability  Styling                                               Pros  \
0          4.5      4.6            \nRoomy cabin\nGreat value\nRefined\n\n   
1          NaN      NaN  \nAcura quality\nAttractive pricing\nPlenty of...   
2          4.0      4.5  \nSuperb build quality\nPlenty of standard fea...   
3          4.5      4.6  \nSpacious cabin\nTech-rich\nRelatively keen p...   
4          4.8      4.8  \nExcellent engines\nRobust resale values\nMan...   

   ...    City Highway Combined Drivetrain      Horsepower          Torque  \
0  ...  19 mpg  26 mpg   22 mpg        FWD  290 @ 6200 RPM  267 @ 4700 rpm   
1  ...  30 mpg  37 mpg   33 mpg        FWD  200 @ 6000 RPM  192 @ 1800 rpm   
2  ...  22 mpg  28 mpg   24 mpg        FWD  272 @ 6500 RPM  280 @ 1600 rpm   
3  ...  19 mpg  26 mpg   22 mpg        FWD  290 @ 6200 RPM  267 @ 4700 rpm   
4  ...  22 mpg  31 mpg   25 mpg        FWD  272 @ 6500 RPM  280 @ 1600 rpm   

                          Engine                  Basic  \
0          V6, i-VTEC, 3.5 Liter  4 years / 50000 miles   
1        4-Cyl, Turbo, 1.5 Liter  4 years / 50000 miles   
2  4-Cyl, VTEC, Turbo, 2.0 Liter  4 years / 50000 miles   
3          V6, i-VTEC, 3.5 Liter  4 years / 50000 miles   
4        4-Cyl, Turbo, 2.0 Liter  4 years / 50000 miles   

              Powertrain                  Corrosion  
0  6 years / 70000 miles  5 years / Unlimited miles  
1  6 years / 70000 miles  5 years / Unlimited miles  
2  6 years / 70000 miles  5 years / Unlimited miles  
3  6 years / 70000 miles  5 years / Unlimited miles  
4  6 years / 70000 miles  5 years / Unlimited miles  

[5 rows x 39 columns]

In [5]:
data.tail()

Unnamed: 0                      Car   Price Value Performance Quality  \
273         274     1992 Volkswagen Golf  $2,165   4.6         4.6     4.6   
274         275   1992 Volkswagen Passat  $2,172   4.4         4.1     4.2   
275         276      1992 Volkswagen Fox  $2,530   4.6         4.4     4.4   
276         277      1992 Volkswagen GTI  $2,789   4.7         4.7     4.6   
277         278  1992 Volkswagen Corrado  $4,507   4.8         4.9     4.9   

     Comfort  Reliability  Styling Pros  ...    City Highway Combined  \
273      4.5          4.6      4.4  NaN  ...  22 mpg  29 mpg   25 mpg   
274      4.4          4.1      4.2  NaN  ...  18 mpg  26 mpg   21 mpg   
275      4.3          4.7      4.2  NaN  ...  22 mpg  30 mpg   25 mpg   
276      4.5          4.8      4.6  NaN  ...  22 mpg  29 mpg   25 mpg   
277      4.8          4.7      4.9  NaN  ...  18 mpg  26 mpg   21 mpg   

    Drivetrain      Horsepower          Torque               Engine  \
273        FWD  100 @ 5400 RPM  107 @ 3400 rpm     4-Cyl, 1.8 Liter   
274        FWD  134 @ 5800 RPM  133 @ 4400 rpm     4-Cyl, 2.0 Liter   
275        FWD   81 @ 5500 RPM   93 @ 3250 rpm     4-Cyl, 1.8 Liter   
276        FWD  105 @ 5400 RPM  110 @ 3400 rpm     4-Cyl, 1.8 Liter   
277        FWD  158 @ 5600 RPM  166 @ 4000 rpm  4-Cyl, Supercharged   

                     Basic             Powertrain                  Corrosion  
273  2 years / 24000 miles  5 years / 50000 miles  6 years / Unlimited miles  
274  2 years / 24000 miles  5 years / 50000 miles  6 years / Unlimited miles  
275  2 years / 24000 miles  5 years / 50000 miles  6 years / Unlimited miles  
276  2 years / 24000 miles  5 years / 50000 miles  6 years / Unlimited miles  
277  2 years / 24000 miles  5 years / 50000 miles  6 years / Unlimited miles  

[5 rows x 39 columns]

### c.Vérifier l'existance des valeurs manquantes

In [6]:
data.isnull().sum()

Unnamed: 0                          0
Car                                 0
Price                               0
Value                             124
Performance                       132
Quality                           133
Comfort                           161
Reliability                       176
Styling                           176
Pros                             4643
Cons                             4643
New                              4643
Heated Steering Wheel            9911
Integrated Garage Door Opener    9806
Interior Ambient Lighting        9902
Curb Weight                       515
Front Head Room                   333
Front Leg Room                    440
Max Seating Capacity               54
Minimum Ground Clearance         3618
Overall Length                    110
Front Shoulder Room               467
Trunk or Cargo Capacity          1772
Turning Diameter                  955
Wheel Base                         81
Width with mirrors               1290
Alloy Wheels

La base contient 9926 lignes,39 colonnes et on remarque que : 
1. Heated Steering Wheel comporte 9911 valeurs manquantes
2. Integrated Garage Door Opener comporte 9806 valeurs manquantes
3. Interior Ambient Lighting comporte 9902 valeurs manquantes
3. LED Headlights comporte 8321 valeurs manquantes

## 2. Statistiques Descriptives

In [7]:
#voir les noms de variables existantes
variables=data.columns
variables

Index(['Unnamed: 0', 'Car', 'Price', 'Value', 'Performance', 'Quality',
       'Comfort', 'Reliability', 'Styling', 'Pros', 'Cons', 'New',
       'Heated Steering Wheel', 'Integrated Garage Door Opener',
       'Interior Ambient Lighting', 'Curb Weight', 'Front Head Room',
       'Front Leg Room', 'Max Seating Capacity', 'Minimum Ground Clearance',
       'Overall Length', 'Front Shoulder Room', 'Trunk or Cargo Capacity',
       'Turning Diameter', 'Wheel Base', 'Width with mirrors', 'Alloy Wheels',
       'Number of Doors', 'LED Headlights', 'City', 'Highway', 'Combined',
       'Drivetrain', 'Horsepower', 'Torque', 'Engine', 'Basic', 'Powertrain',
       'Corrosion'],
      dtype='object')

In [8]:
#afficher le type de chaque variable
data.dtypes

Unnamed: 0                         int64
Car                               object
Price                             object
Value                             object
Performance                       object
Quality                           object
Comfort                          float64
Reliability                      float64
Styling                          float64
Pros                              object
Cons                              object
New                               object
Heated Steering Wheel             object
Integrated Garage Door Opener     object
Interior Ambient Lighting         object
Curb Weight                       object
Front Head Room                   object
Front Leg Room                    object
Max Seating Capacity             float64
Minimum Ground Clearance          object
Overall Length                    object
Front Shoulder Room               object
Trunk or Cargo Capacity           object
Turning Diameter                  object
Wheel Base      

In [9]:
data.describe()

Unnamed: 0      Comfort  Reliability      Styling  \
count  9926.000000  9765.000000  9750.000000  9750.000000   
mean    253.763651     4.477634     4.494605     4.520810   
std     238.611363     0.308532     0.321978     0.278797   
min       1.000000     1.000000     2.000000     2.800000   
25%      71.000000     4.300000     4.400000     4.400000   
50%     171.000000     4.500000     4.600000     4.600000   
75%     378.000000     4.700000     4.700000     4.700000   
max    1075.000000     5.000000     5.000000     5.000000   

       Max Seating Capacity  
count           9872.000000  
mean               5.220421  
std                2.006735  
min                0.000000  
25%                5.000000  
50%                5.000000  
75%                6.000000  
max               15.000000

In [10]:
#voir les différentes valeurs de la variable 'Alloy Wheels'
data['Alloy Wheels'].unique()

array(['Available', nan], dtype=object)

In [11]:
#voir les différentes valeurs de la variable Integrated Garage Door Opener'
data['Integrated Garage Door Opener'].unique()

array([nan, 'Port Installed'], dtype=object)

On peut remarquer pour les variables 'Alloy Wheels' et 'LED Headlights' que les cases vides correspondent à Not Available
de même pour la variable 'Integrated Garage Door Opener' les cases vides correspondent à Port Not Installed

## 3. Tranformation et Nettoyage de la base :

### a. Données manquantes

**Alloy Wheels**

In [12]:
#Transformation des colo
data['Alloy Wheels']=data['Alloy Wheels'].fillna('Not Available')

In [13]:
data['Alloy Wheels'].unique()

array(['Available', 'Not Available'], dtype=object)

In [14]:
#changer le nom de la colonne pour faciliter le travail
data=data.rename(columns={'Alloy Wheels': 'alloy_wheels'})
data['alloy_wheels']

0      Available
1      Available
2      Available
3      Available
4      Available
         ...    
273    Available
274    Available
275    Available
276    Available
277    Available
Name: alloy_wheels, Length: 9926, dtype: object

 **LED Headlights**

In [15]:
data['LED Headlights']=data['LED Headlights'].fillna('Not Available')

In [16]:
data['LED Headlights'].unique()

array(['Available', 'Not Available'], dtype=object)

In [17]:
#changer le nom de la colonne pour faciliter le travail
data=data.rename(columns={'LED Headlights': 'led_headlights'})
data['led_headlights']

0          Available
1          Available
2          Available
3          Available
4          Available
           ...      
273    Not Available
274    Not Available
275    Not Available
276    Not Available
277    Not Available
Name: led_headlights, Length: 9926, dtype: object

**Integrated Garage Door Opener**

In [18]:
data['Integrated Garage Door Opener']=data['Integrated Garage Door Opener'].fillna('Port Not Installed')

In [19]:
data['Integrated Garage Door Opener'].unique()

array(['Port Not Installed', 'Port Installed'], dtype=object)

In [20]:
#changer le nom de la colonne pour faciliter le travail
data=data.rename(columns={'Integrated Garage Door Opener': 'integrated_garage_door_opener'})
data['integrated_garage_door_opener']

0      Port Not Installed
1      Port Not Installed
2      Port Not Installed
3      Port Not Installed
4      Port Not Installed
              ...        
273    Port Not Installed
274    Port Not Installed
275    Port Not Installed
276    Port Not Installed
277    Port Not Installed
Name: integrated_garage_door_opener, Length: 9926, dtype: object

**Ratings**

In [21]:
# suppression des lignes qui ne contiennent pas les ratings des variables 'Comfort', 'Performance', 'Quality', 'Value', 'Reliability', 'Styling'

columns_to_check=['Comfort', 'Performance', 'Quality', 'Value', 'Reliability', 'Styling','Engine','Basic','Powertrain']
data_cleaned_1= data.dropna(subset=columns_to_check)

#suppression des colonnes qui contiennent plusieurs valeurs manquantes
columns_to_drop=['Unnamed: 0','Heated Steering Wheel','Interior Ambient Lighting','Minimum Ground Clearance','Corrosion','City','Pros','Cons','New']
data_cleaned_2= data_cleaned_1.drop(columns=columns_to_drop)
data_cleaned_2.isnull().sum()

Car                                 0
Price                               0
Value                               0
Performance                         0
Quality                             0
Comfort                             0
Reliability                         0
Styling                             0
integrated_garage_door_opener       0
Curb Weight                       453
Front Head Room                   267
Front Leg Room                    374
Max Seating Capacity               21
Overall Length                     76
Front Shoulder Room               397
Trunk or Cargo Capacity          1729
Turning Diameter                  913
Wheel Base                         52
Width with mirrors               1230
alloy_wheels                        0
Number of Doors                     0
led_headlights                      0
Highway                          1431
Combined                         1338
Drivetrain                          0
Horsepower                        453
Torque      

**Pour la colonne 'Basic'**  

la colonne 'Basic' contient la durée de la couverture en années et la limite de kilométrage ou la distance maximale parcourue par le véhicule pendant la période de garantie ou d'entretien en deux colonnes numériques des équipements et fonctionnalités essentiels qui sont fournis en standard, sans ajouts ni options supplémentaires.

In [22]:
data_cleaned_2[['max_years_warrantly', 'max_miles_warrantly']] = data_cleaned_2['Basic'].str.split('/', expand=True)
data_cleaned_2['max_years_warrantly'] = data_cleaned_2['max_years_warrantly'].str.strip().str.replace('years', '')
data_cleaned_2['max_miles_warrantly'] = data_cleaned_2['max_miles_warrantly'].str.strip().str.replace('miles', '')
data_cleaned_2['max_years_warrantly'] = pd.to_numeric(data_cleaned_2['max_years_warrantly'], errors='coerce')
data_cleaned_2['max_miles_warrantly'] = pd.to_numeric(data_cleaned_2['max_miles_warrantly'], errors='coerce')
data_cleaned_2=data_cleaned_2.drop('Basic', axis=1)

**Pour la colonne 'Powertrain'**  

la colonne 'Powertrain' contient la durée de la couverture en années et la limite de kilométrage ou la distance maximale parcourue par le véhicule pendant la période de garantie ou d'entretien en deux colonnes numériques des composants qui génèrent et transmettent la puissance nécessaire pour faire avancer le véhicule.

In [23]:
data_cleaned_2[['max_years_powertrain', 'max_miles_powertrain']] = data_cleaned_2['Powertrain'].str.split('/', expand=True)
data_cleaned_2['max_years_powertrain'] = data_cleaned_2['max_years_powertrain'].str.strip().str.replace('years', '')
data_cleaned_2['max_miles_powertrain'] = data_cleaned_2['max_miles_powertrain'].str.strip().str.replace('miles', '')
data_cleaned_2['max_years_powertrain'] = pd.to_numeric(data_cleaned_2['max_years_powertrain'], errors='coerce')
data_cleaned_2['max_miles_powertrain'] = pd.to_numeric(data_cleaned_2['max_miles_powertrain'], errors='coerce')
data_cleaned_2=data_cleaned_2.drop('Powertrain', axis=1)
data_cleaned_2

Car    Price Value Performance Quality  Comfort  \
0             2024 Acura MDX  $51,045   4.3         4.3     4.4      4.6   
2             2023 Acura RDX  $42,945   3.9         4.3     4.0      4.3   
3             2023 Acura MDX  $50,745   4.3         4.3     4.4      4.6   
4             2023 Acura TLX  $41,245   4.5         4.5     4.7      4.4   
6             2022 Acura RDX  $41,795   3.9         4.3     4.0      4.3   
..                       ...      ...   ...         ...     ...      ...   
273     1992 Volkswagen Golf   $2,165   4.6         4.6     4.6      4.5   
274   1992 Volkswagen Passat   $2,172   4.4         4.1     4.2      4.4   
275      1992 Volkswagen Fox   $2,530   4.6         4.4     4.4      4.3   
276      1992 Volkswagen GTI   $2,789   4.7         4.7     4.6      4.5   
277  1992 Volkswagen Corrado   $4,507   4.8         4.9     4.9      4.8   

     Reliability  Styling integrated_garage_door_opener Curb Weight  ...  \
0            4.5      4.6            Port Not Installed   4271 lbs.  ...   
2            4.0      4.5            Port Not Installed   3788 lbs.  ...   
3            4.5      4.6            Port Not Installed   4271 lbs.  ...   
4            4.8      4.8            Port Not Installed   3709 lbs.  ...   
6            4.0      4.5            Port Not Installed   3788 lbs.  ...   
..           ...      ...                           ...         ...  ...   
273          4.6      4.4            Port Not Installed   2320 lbs.  ...   
274          4.1      4.2            Port Not Installed   2985 lbs.  ...   
275          4.7      4.2            Port Not Installed   2172 lbs.  ...   
276          4.8      4.6            Port Not Installed   2346 lbs.  ...   
277          4.7      4.9            Port Not Installed   2675 lbs.  ...   

    Highway Combined  Drivetrain      Horsepower          Torque  \
0    26 mpg   22 mpg         FWD  290 @ 6200 RPM  267 @ 4700 rpm   
2    28 mpg   24 mpg         FWD  272 @ 6500 RPM  280 @ 1600 rpm   
3    26 mpg   22 mpg         FWD  290 @ 6200 RPM  267 @ 4700 rpm   
4    31 mpg   25 mpg         FWD  272 @ 6500 RPM  280 @ 1600 rpm   
6    28 mpg   24 mpg         FWD  272 @ 6500 RPM  280 @ 1600 rpm   
..      ...      ...         ...             ...             ...   
273  29 mpg   25 mpg         FWD  100 @ 5400 RPM  107 @ 3400 rpm   
274  26 mpg   21 mpg         FWD  134 @ 5800 RPM  133 @ 4400 rpm   
275  30 mpg   25 mpg         FWD   81 @ 5500 RPM   93 @ 3250 rpm   
276  29 mpg   25 mpg         FWD  105 @ 5400 RPM  110 @ 3400 rpm   
277  26 mpg   21 mpg         FWD  158 @ 5600 RPM  166 @ 4000 rpm   

                            Engine max_years_warrantly max_miles_warrantly  \
0            V6, i-VTEC, 3.5 Liter                   4             50000.0   
2    4-Cyl, VTEC, Turbo, 2.0 Liter                   4             50000.0   
3            V6, i-VTEC, 3.5 Liter                   4             50000.0   
4          4-Cyl, Turbo, 2.0 Liter                   4             50000.0   
6    4-Cyl, VTEC, Turbo, 2.0 Liter                   4             50000.0   
..                             ...                 ...                 ...   
273               4-Cyl, 1.8 Liter                   2             24000.0   
274               4-Cyl, 2.0 Liter                   2             24000.0   
275               4-Cyl, 1.8 Liter                   2             24000.0   
276               4-Cyl, 1.8 Liter                   2             24000.0   
277            4-Cyl, Supercharged                   2             24000.0   

    max_years_powertrain max_miles_powertrain  
0                    6.0              70000.0  
2                    6.0              70000.0  
3                    6.0              70000.0  
4                    6.0              70000.0  
6                    6.0              70000.0  
..                   ...                  ...  
273                  5.0              50000.0  
274                  5.0              50000.0  
275          

In [24]:
data_cleaned_2.isnull().sum()

Car                                 0
Price                               0
Value                               0
Performance                         0
Quality                             0
Comfort                             0
Reliability                         0
Styling                             0
integrated_garage_door_opener       0
Curb Weight                       453
Front Head Room                   267
Front Leg Room                    374
Max Seating Capacity               21
Overall Length                     76
Front Shoulder Room               397
Trunk or Cargo Capacity          1729
Turning Diameter                  913
Wheel Base                         52
Width with mirrors               1230
alloy_wheels                        0
Number of Doors                     0
led_headlights                      0
Highway                          1431
Combined                         1338
Drivetrain                          0
Horsepower                        453
Torque      

In [25]:
columns=['Max Seating Capacity','max_years_warrantly','max_miles_warrantly','max_years_powertrain','max_miles_powertrain']
data_cleaned_3=data_cleaned_2.dropna(subset=columns)
data_cleaned_3.isnull().sum()

Car                                 0
Price                               0
Value                               0
Performance                         0
Quality                             0
Comfort                             0
Reliability                         0
Styling                             0
integrated_garage_door_opener       0
Curb Weight                       445
Front Head Room                   263
Front Leg Room                    370
Max Seating Capacity                0
Overall Length                     73
Front Shoulder Room               392
Trunk or Cargo Capacity          1726
Turning Diameter                  906
Wheel Base                         50
Width with mirrors               1226
alloy_wheels                        0
Number of Doors                     0
led_headlights                      0
Highway                          1426
Combined                         1334
Drivetrain                          0
Horsepower                        452
Torque      

**La colonne Year**

Il s'agit d'une nouvelle colonne qui contient l'année de fabrication de la voiture et c'est à partir de la colonne Car qui contient le nom de la voiture complet :

In [26]:
#Ajout de la colonne year à partir de la colonne Car
year=[]
for c in data_cleaned_3.Car :
    year.append(int(c.split(' ')[0]))

data_cleaned_3['Year']=year
data_cleaned_3

<ipython-input-26-6443074b2c2a>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_cleaned_3['Year']=year


Car    Price Value Performance Quality  Comfort  \
0             2024 Acura MDX  $51,045   4.3         4.3     4.4      4.6   
2             2023 Acura RDX  $42,945   3.9         4.3     4.0      4.3   
3             2023 Acura MDX  $50,745   4.3         4.3     4.4      4.6   
4             2023 Acura TLX  $41,245   4.5         4.5     4.7      4.4   
6             2022 Acura RDX  $41,795   3.9         4.3     4.0      4.3   
..                       ...      ...   ...         ...     ...      ...   
273     1992 Volkswagen Golf   $2,165   4.6         4.6     4.6      4.5   
274   1992 Volkswagen Passat   $2,172   4.4         4.1     4.2      4.4   
275      1992 Volkswagen Fox   $2,530   4.6         4.4     4.4      4.3   
276      1992 Volkswagen GTI   $2,789   4.7         4.7     4.6      4.5   
277  1992 Volkswagen Corrado   $4,507   4.8         4.9     4.9      4.8   

     Reliability  Styling integrated_garage_door_opener Curb Weight  ...  \
0            4.5      4.6            Port Not Installed   4271 lbs.  ...   
2            4.0      4.5            Port Not Installed   3788 lbs.  ...   
3            4.5      4.6            Port Not Installed   4271 lbs.  ...   
4            4.8      4.8            Port Not Installed   3709 lbs.  ...   
6            4.0      4.5            Port Not Installed   3788 lbs.  ...   
..           ...      ...                           ...         ...  ...   
273          4.6      4.4            Port Not Installed   2320 lbs.  ...   
274          4.1      4.2            Port Not Installed   2985 lbs.  ...   
275          4.7      4.2            Port Not Installed   2172 lbs.  ...   
276          4.8      4.6            Port Not Installed   2346 lbs.  ...   
277          4.7      4.9            Port Not Installed   2675 lbs.  ...   

    Combined Drivetrain      Horsepower          Torque  \
0     22 mpg        FWD  290 @ 6200 RPM  267 @ 4700 rpm   
2     24 mpg        FWD  272 @ 6500 RPM  280 @ 1600 rpm   
3     22 mpg        FWD  290 @ 6200 RPM  267 @ 4700 rpm   
4     25 mpg        FWD  272 @ 6500 RPM  280 @ 1600 rpm   
6     24 mpg        FWD  272 @ 6500 RPM  280 @ 1600 rpm   
..       ...        ...             ...             ...   
273   25 mpg        FWD  100 @ 5400 RPM  107 @ 3400 rpm   
274   21 mpg        FWD  134 @ 5800 RPM  133 @ 4400 rpm   
275   25 mpg        FWD   81 @ 5500 RPM   93 @ 3250 rpm   
276   25 mpg        FWD  105 @ 5400 RPM  110 @ 3400 rpm   
277   21 mpg        FWD  158 @ 5600 RPM  166 @ 4000 rpm   

                            Engine max_years_warrantly max_miles_warrantly  \
0            V6, i-VTEC, 3.5 Liter                   4             50000.0   
2    4-Cyl, VTEC, Turbo, 2.0 Liter                   4             50000.0   
3            V6, i-VTEC, 3.5 Liter                   4             50000.0   
4          4-Cyl, Turbo, 2.0 Liter                   4             50000.0   
6    4-Cyl, VTEC, Turbo, 2.0 Liter                   4             50000.0   
..                             ...                 ...                 ...   
273               4-Cyl, 1.8 Liter                   2             24000.0   
274               4-Cyl, 2.0 Liter                   2             24000.0   
275               4-Cyl, 1.8 Liter                   2             24000.0   
276               4-Cyl, 1.8 Liter                   2             24000.0   
277            4-Cyl, Supercharged                   2             24000.0   

    max_years_powertrain max_miles_powertrain  Year  
0                    6.0              70000.0  2024  
2                    6.0              70000.0  2023  
3                    6.0              70000.0  2023  
4                    6.0              70000.0  2023  
6                    6.0              70000.0  2022  
..                   ...                  ...   ...  
273                  5.0              50000.0  1992  
274                  5.0              50000.0  1992  
275                  5.0              50000.0  1992  
276          

**La colonne Brand**

Il s'agit d'une nouvelle colonne qui contient le modèle de la voiture et c'est à partir de la colonne Car qui contient le nom de la voiture complet :

In [27]:
brand=[]
for c in data_cleaned_3.Car :
    brand.append(c.split(' ')[1])
    
data_cleaned_3['Brand']=brand

data_cleaned_3

<ipython-input-27-46b2e6392e64>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_cleaned_3['Brand']=brand


Car    Price Value Performance Quality  Comfort  \
0             2024 Acura MDX  $51,045   4.3         4.3     4.4      4.6   
2             2023 Acura RDX  $42,945   3.9         4.3     4.0      4.3   
3             2023 Acura MDX  $50,745   4.3         4.3     4.4      4.6   
4             2023 Acura TLX  $41,245   4.5         4.5     4.7      4.4   
6             2022 Acura RDX  $41,795   3.9         4.3     4.0      4.3   
..                       ...      ...   ...         ...     ...      ...   
273     1992 Volkswagen Golf   $2,165   4.6         4.6     4.6      4.5   
274   1992 Volkswagen Passat   $2,172   4.4         4.1     4.2      4.4   
275      1992 Volkswagen Fox   $2,530   4.6         4.4     4.4      4.3   
276      1992 Volkswagen GTI   $2,789   4.7         4.7     4.6      4.5   
277  1992 Volkswagen Corrado   $4,507   4.8         4.9     4.9      4.8   

     Reliability  Styling integrated_garage_door_opener Curb Weight  ...  \
0            4.5      4.6            Port Not Installed   4271 lbs.  ...   
2            4.0      4.5            Port Not Installed   3788 lbs.  ...   
3            4.5      4.6            Port Not Installed   4271 lbs.  ...   
4            4.8      4.8            Port Not Installed   3709 lbs.  ...   
6            4.0      4.5            Port Not Installed   3788 lbs.  ...   
..           ...      ...                           ...         ...  ...   
273          4.6      4.4            Port Not Installed   2320 lbs.  ...   
274          4.1      4.2            Port Not Installed   2985 lbs.  ...   
275          4.7      4.2            Port Not Installed   2172 lbs.  ...   
276          4.8      4.6            Port Not Installed   2346 lbs.  ...   
277          4.7      4.9            Port Not Installed   2675 lbs.  ...   

    Drivetrain      Horsepower          Torque                         Engine  \
0          FWD  290 @ 6200 RPM  267 @ 4700 rpm          V6, i-VTEC, 3.5 Liter   
2          FWD  272 @ 6500 RPM  280 @ 1600 rpm  4-Cyl, VTEC, Turbo, 2.0 Liter   
3          FWD  290 @ 6200 RPM  267 @ 4700 rpm          V6, i-VTEC, 3.5 Liter   
4          FWD  272 @ 6500 RPM  280 @ 1600 rpm        4-Cyl, Turbo, 2.0 Liter   
6          FWD  272 @ 6500 RPM  280 @ 1600 rpm  4-Cyl, VTEC, Turbo, 2.0 Liter   
..         ...             ...             ...                            ...   
273        FWD  100 @ 5400 RPM  107 @ 3400 rpm               4-Cyl, 1.8 Liter   
274        FWD  134 @ 5800 RPM  133 @ 4400 rpm               4-Cyl, 2.0 Liter   
275        FWD   81 @ 5500 RPM   93 @ 3250 rpm               4-Cyl, 1.8 Liter   
276        FWD  105 @ 5400 RPM  110 @ 3400 rpm               4-Cyl, 1.8 Liter   
277        FWD  158 @ 5600 RPM  166 @ 4000 rpm            4-Cyl, Supercharged   

    max_years_warrantly max_miles_warrantly max_years_powertrain  \
0                     4             50000.0                  6.0   
2                     4             50000.0                  6.0   
3                     4             50000.0                  6.0   
4                     4             50000.0                  6.0   
6                     4             50000.0                  6.0   
..                  ...                 ...                  ...   
273                   2             24000.0                  5.0   
274                   2             24000.0                  5.0   
275                   2             24000.0                  5.0   
276                   2             24000.0                  5.0   
277                   2             24000.0                  5.0   

    max_miles_powertrain  Year       Brand  
0                70000.0  2024       Acura  
2                70000.0  2023       Acura  
3                70000.0  2023       Acura  
4                70000.0  2023       Acura  
6                70000.0  2022       Acura  
..                   ...   ...         ...  
273              50000.0  1992  Volkswagen  
274              50000.0  1992  Volkswagen  
275 

**Tunk and cargo capacity**

cette variable nous indique la taille du malle :

In [28]:
#changer le nom de la colonne pour faciliter le travail
data_cleaned_3=data_cleaned_3.rename(columns={'Trunk or Cargo Capacity': 'trunk_capacity'})
data_cleaned_3['trunk_capacity']

0      95.0 cu.ft.
2      29.5 cu.ft.
3      95.0 cu.ft.
4      13.5 cu.ft.
6      29.5 cu.ft.
          ...     
273    17.8 cu.ft.
274    14.2 cu.ft.
275     9.9 cu.ft.
276    17.8 cu.ft.
277    18.6 cu.ft.
Name: trunk_capacity, Length: 9713, dtype: object

In [29]:
trunk=[]
import math
for c in data_cleaned_3.trunk_capacity :
    if str(c).lower()=="nan":
         trunk.append(100.0)
    else :     
        c1=c[:-7]
        if c1[-1]==' ' :
            trunk.append(float(c1[:-1]))
        else :
            trunk.append(float(c1))        
    
data_cleaned_3['trunk_cap']=trunk

data_cleaned_3

Car    Price Value Performance Quality  Comfort  \
0             2024 Acura MDX  $51,045   4.3         4.3     4.4      4.6   
2             2023 Acura RDX  $42,945   3.9         4.3     4.0      4.3   
3             2023 Acura MDX  $50,745   4.3         4.3     4.4      4.6   
4             2023 Acura TLX  $41,245   4.5         4.5     4.7      4.4   
6             2022 Acura RDX  $41,795   3.9         4.3     4.0      4.3   
..                       ...      ...   ...         ...     ...      ...   
273     1992 Volkswagen Golf   $2,165   4.6         4.6     4.6      4.5   
274   1992 Volkswagen Passat   $2,172   4.4         4.1     4.2      4.4   
275      1992 Volkswagen Fox   $2,530   4.6         4.4     4.4      4.3   
276      1992 Volkswagen GTI   $2,789   4.7         4.7     4.6      4.5   
277  1992 Volkswagen Corrado   $4,507   4.8         4.9     4.9      4.8   

     Reliability  Styling integrated_garage_door_opener Curb Weight  ...  \
0            4.5      4.6            Port Not Installed   4271 lbs.  ...   
2            4.0      4.5            Port Not Installed   3788 lbs.  ...   
3            4.5      4.6            Port Not Installed   4271 lbs.  ...   
4            4.8      4.8            Port Not Installed   3709 lbs.  ...   
6            4.0      4.5            Port Not Installed   3788 lbs.  ...   
..           ...      ...                           ...         ...  ...   
273          4.6      4.4            Port Not Installed   2320 lbs.  ...   
274          4.1      4.2            Port Not Installed   2985 lbs.  ...   
275          4.7      4.2            Port Not Installed   2172 lbs.  ...   
276          4.8      4.6            Port Not Installed   2346 lbs.  ...   
277          4.7      4.9            Port Not Installed   2675 lbs.  ...   

         Horsepower          Torque                         Engine  \
0    290 @ 6200 RPM  267 @ 4700 rpm          V6, i-VTEC, 3.5 Liter   
2    272 @ 6500 RPM  280 @ 1600 rpm  4-Cyl, VTEC, Turbo, 2.0 Liter   
3    290 @ 6200 RPM  267 @ 4700 rpm          V6, i-VTEC, 3.5 Liter   
4    272 @ 6500 RPM  280 @ 1600 rpm        4-Cyl, Turbo, 2.0 Liter   
6    272 @ 6500 RPM  280 @ 1600 rpm  4-Cyl, VTEC, Turbo, 2.0 Liter   
..              ...             ...                            ...   
273  100 @ 5400 RPM  107 @ 3400 rpm               4-Cyl, 1.8 Liter   
274  134 @ 5800 RPM  133 @ 4400 rpm               4-Cyl, 2.0 Liter   
275   81 @ 5500 RPM   93 @ 3250 rpm               4-Cyl, 1.8 Liter   
276  105 @ 5400 RPM  110 @ 3400 rpm               4-Cyl, 1.8 Liter   
277  158 @ 5600 RPM  166 @ 4000 rpm            4-Cyl, Supercharged   

    max_years_warrantly max_miles_warrantly max_years_powertrain  \
0                     4             50000.0                  6.0   
2                     4             50000.0                  6.0   
3                     4             50000.0                  6.0   
4                     4             50000.0                  6.0   
6                     4             50000.0                  6.0   
..                  ...                 ...                  ...   
273                   2             24000.0                  5.0   
274                   2             24000.0                  5.0   
275                   2             24000.0                  5.0   
276                   2             24000.0                  5.0   
277                   2             24000.0                  5.0   

    max_miles_powertrain  Year       Brand trunk_cap  
0                70000.0  2024       Acura      95.0  
2                70000.0  2023       Acura      29.5  
3                70000.0  2023       Acura      95.0  
4                70000.0  2023       Acura      13.5  
6                70000.0  2022       Acura      29.5  
..                   ...   ...         ...       ...  
273              50000.0  1992  Volkswagen      17.8  
274              50000.0  1992  Volkswagen      14.2  
275              50000.0  1992  Volkswagen    

Je vais transformer la variable trunk_cap en une variable catégorique :

**1. Très Petite (Micro/Compacte) :**

Cargo capacity < 8 cubic feet

**2. Petite (Sous-compacte/Compacte) :**

8 <= Cargo capacity < 12 cubic feet

**3. Modérée (Taille intermédiaire) :**

12 <= Cargo capacity < 20 cubic feet

**4. Spacieuse (VUS de grande taille/Tailles intermédiaires, Breaks) :**

20 <= Cargo capacity < 30 cubic feet

**5. Très Spacieuse (VUS de grande taille, Minifourgonnettes) :**

Cargo capacity >= 30 cubic feet 

In [30]:
# Définir les intervalles et les étiquettes correspondantes
intervals = [float('-inf'), 8, 12, 20, 30, float('inf')]
labels = ['Very Small', 'Small', 'Moderate', 'Spacious', 'Very Spacious']

# Convertir la colonne 'trunk_cap' en une colonne catégorique basée sur les intervalles et les étiquettes
data_cleaned_3['trunk_cap_categ'] = pd.cut(data_cleaned_3['trunk_cap'], bins=intervals, labels=labels, right=False)

In [31]:
data_cleaned_3

Car    Price Value Performance Quality  Comfort  \
0             2024 Acura MDX  $51,045   4.3         4.3     4.4      4.6   
2             2023 Acura RDX  $42,945   3.9         4.3     4.0      4.3   
3             2023 Acura MDX  $50,745   4.3         4.3     4.4      4.6   
4             2023 Acura TLX  $41,245   4.5         4.5     4.7      4.4   
6             2022 Acura RDX  $41,795   3.9         4.3     4.0      4.3   
..                       ...      ...   ...         ...     ...      ...   
273     1992 Volkswagen Golf   $2,165   4.6         4.6     4.6      4.5   
274   1992 Volkswagen Passat   $2,172   4.4         4.1     4.2      4.4   
275      1992 Volkswagen Fox   $2,530   4.6         4.4     4.4      4.3   
276      1992 Volkswagen GTI   $2,789   4.7         4.7     4.6      4.5   
277  1992 Volkswagen Corrado   $4,507   4.8         4.9     4.9      4.8   

     Reliability  Styling integrated_garage_door_opener Curb Weight  ...  \
0            4.5      4.6            Port Not Installed   4271 lbs.  ...   
2            4.0      4.5            Port Not Installed   3788 lbs.  ...   
3            4.5      4.6            Port Not Installed   4271 lbs.  ...   
4            4.8      4.8            Port Not Installed   3709 lbs.  ...   
6            4.0      4.5            Port Not Installed   3788 lbs.  ...   
..           ...      ...                           ...         ...  ...   
273          4.6      4.4            Port Not Installed   2320 lbs.  ...   
274          4.1      4.2            Port Not Installed   2985 lbs.  ...   
275          4.7      4.2            Port Not Installed   2172 lbs.  ...   
276          4.8      4.6            Port Not Installed   2346 lbs.  ...   
277          4.7      4.9            Port Not Installed   2675 lbs.  ...   

             Torque                         Engine  max_years_warrantly  \
0    267 @ 4700 rpm          V6, i-VTEC, 3.5 Liter                    4   
2    280 @ 1600 rpm  4-Cyl, VTEC, Turbo, 2.0 Liter                    4   
3    267 @ 4700 rpm          V6, i-VTEC, 3.5 Liter                    4   
4    280 @ 1600 rpm        4-Cyl, Turbo, 2.0 Liter                    4   
6    280 @ 1600 rpm  4-Cyl, VTEC, Turbo, 2.0 Liter                    4   
..              ...                            ...                  ...   
273  107 @ 3400 rpm               4-Cyl, 1.8 Liter                    2   
274  133 @ 4400 rpm               4-Cyl, 2.0 Liter                    2   
275   93 @ 3250 rpm               4-Cyl, 1.8 Liter                    2   
276  110 @ 3400 rpm               4-Cyl, 1.8 Liter                    2   
277  166 @ 4000 rpm            4-Cyl, Supercharged                    2   

    max_miles_warrantly max_years_powertrain max_miles_powertrain  Year  \
0               50000.0                  6.0              70000.0  2024   
2               50000.0                  6.0              70000.0  2023   
3               50000.0                  6.0              70000.0  2023   
4               50000.0                  6.0              70000.0  2023   
6               50000.0                  6.0              70000.0  2022   
..                  ...                  ...                  ...   ...   
273             24000.0                  5.0              50000.0  1992   
274             24000.0                  5.0              50000.0  1992   
275             24000.0                  5.0              50000.0  1992   
276             24000.0                  5.0              50000.0  1992   
277             24000.0                  5.0              50000.0  1992   

          Brand trunk_cap trunk_cap_categ  
0         Acura      95.0   Very Spacious  
2         Acura      29.5        Spacious  
3         Acura      95.0   Very Spacious  
4         Acura      13.5        Moderate  
6         Acura      29.5        Spacious  
..          ...       ...             ...  
273  Volkswagen      17.8        Moderate  
274  Volkswagen      14.2        Moderate  
2

 **Number of Doors**

In [32]:
#changer le nom de la colonne pour faciliter le travail
data_cleaned_3=data_cleaned_3.rename(columns={'Number of Doors': 'nb_doors'})
data_cleaned_3['nb_doors']

0      4 doors
2      4 doors
3      4 doors
4      4 doors
6      4 doors
        ...   
273    2 doors
274    4 doors
275    2 doors
276    2 doors
277    2 doors
Name: nb_doors, Length: 9713, dtype: object

In [33]:
doors=[]
for c in data_cleaned_3.nb_doors :
    doors.append(c.split(' ')[0])
    
data_cleaned_3['nb_doors']=doors

data_cleaned_3['nb_doors']

0      4
2      4
3      4
4      4
6      4
      ..
273    2
274    4
275    2
276    2
277    2
Name: nb_doors, Length: 9713, dtype: object

### **Drivetrain**

In [34]:
data_cleaned_3['Drivetrain'].unique()

array(['FWD', 'AWD', '2WD', '4WD', 'RWD'], dtype=object)


**FWD (Front-Wheel Drive) :** 

Dans un véhicule à traction avant (FWD), la puissance du moteur est transmise aux roues avant. Cette configuration est courante dans de nombreuses voitures plus petites et offre une bonne traction dans la plupart des conditions.

**AWD (All-Wheel Drive) :**

La transmission intégrale (AWD) signifie que la puissance est distribuée aux quatre roues du véhicule. Elle peut être soit à temps plein, où toutes les roues sont propulsées en permanence, soit à temps partiel, où la puissance est envoyée aux roues qui en ont le plus besoin en fonction des conditions de la route.

**2WD (Two-Wheel Drive) :**

Ce terme est souvent utilisé de manière interchangeable avec 2RM (2 roues motrices), et il se réfère généralement aux véhicules à traction avant (FWD) ou à propulsion arrière (RWD).

**4WD (Four-Wheel Drive) :**

La transmission à quatre roues motrices (4WD) signifie que les quatre roues sont propulsées, généralement utilisée dans les véhicules tout-terrain ou les camions pour une meilleure traction dans des terrains difficiles. Certains systèmes 4WD peuvent être activés ou désactivés en fonction des conditions.

**RWD (Rear-Wheel Drive) :**

Dans un véhicule à propulsion arrière (RWD), la puissance du moteur est transmise aux roues arrière. Cette configuration est souvent utilisée dans les voitures de sport et les véhicules plus grands.

### **Engine**

Exemples :

**V6 (6-Cylinder) :**
Cela fait référence au nombre de cylindres dans le moteur. Dans ce cas, il s'agit d'un moteur à six cylindres. Les moteurs à six cylindres, comme les moteurs V6, sont connus pour leur puissance et leurs performances accrues par rapport aux moteurs à quatre cylindres.

**4-Cyl (4-Cylinder) :** 
Cela fait référence au nombre de cylindres dans le moteur. Les moteurs peuvent avoir différents nombres de cylindres, et dans ce cas, il s'agit d'un moteur à quatre cylindres.

In [45]:
eng=[]
for c in data_cleaned_3.Engine :
    if ',' in c :
        eng.append(c.split(',')[0])
    elif '.' in c :
        eng.append(c.split('.')[0])
    elif '  '  in c :
        eng.append(c.split('  ')[0])
    else : 
        eng.append(c.split(' ')[0])
    
data_cleaned_3['Engine']=eng

data_cleaned_3['Engine']

0         V6
2      4-Cyl
3         V6
4      4-Cyl
6      4-Cyl
       ...  
273    4-Cyl
274    4-Cyl
275    4-Cyl
276    4-Cyl
277    4-Cyl
Name: Engine, Length: 9713, dtype: object

### **Création de la colonne Model**

In [46]:
model=[]
for c in data_cleaned_3.Car :
    words=c.split(' ')
    model.append(words[-1])

data_cleaned_3['Model']=model
data_cleaned_3['Model']

0          MDX
2          RDX
3          MDX
4          TLX
6          RDX
        ...   
273       Golf
274     Passat
275        Fox
276        GTI
277    Corrado
Name: Model, Length: 9713, dtype: object

**Price**

Je vais transformer le type de la colonne Price de "object" à "float" :

In [47]:
price=[]
for p in data_cleaned_3.Price :
    price.append(p[1:])
    
data_cleaned_3['Price']=price
data_cleaned_3['Price'] = data_cleaned_3['Price'].str.replace(',', '').astype(float)

data_cleaned_3['Price']

TypeError: 'float' object is not subscriptable

In [ ]:
data_cleaned_3['Value'] = data_cleaned_3['Value'].astype(float)
data_cleaned_3['Value']

In [ ]:
data_cleaned_3['Quality'] = data_cleaned_3['Quality'].astype(float)
data_cleaned_3['Quality'] 

In [48]:
#changer le nom de la colonne pour faciliter le travail
data_cleaned_3=data_cleaned_3.rename(columns={'Max Seating Capacity': 'max_seating_capacity'})
data_cleaned_3['max_seating_capacity']

0      7.0
2      5.0
3      7.0
4      5.0
6      5.0
      ... 
273    5.0
274    5.0
275    4.0
276    5.0
277    4.0
Name: max_seating_capacity, Length: 9713, dtype: float64

In [49]:
#Création de la nouvelle base en recopiant des colonnes de data_cleaned_3 
selected_columns=['Car', 'Price','Comfort', 'Performance', 'Quality', 'Value', 'Reliability', 'Styling','integrated_garage_door_opener','max_seating_capacity','alloy_wheels','nb_doors','led_headlights','Drivetrain','Engine','max_years_warrantly','max_miles_warrantly','max_years_powertrain','max_miles_powertrain','trunk_cap_categ','Year','Brand','Model']
data_cleaned = data_cleaned_3.loc[:,selected_columns].copy()
data_cleaned 

Car    Price  Comfort Performance  Quality  Value  \
0             2024 Acura MDX  51045.0      4.6         4.3      4.4    4.3   
2             2023 Acura RDX  42945.0      4.3         4.3      4.0    3.9   
3             2023 Acura MDX  50745.0      4.6         4.3      4.4    4.3   
4             2023 Acura TLX  41245.0      4.4         4.5      4.7    4.5   
6             2022 Acura RDX  41795.0      4.3         4.3      4.0    3.9   
..                       ...      ...      ...         ...      ...    ...   
273     1992 Volkswagen Golf   2165.0      4.5         4.6      4.6    4.6   
274   1992 Volkswagen Passat   2172.0      4.4         4.1      4.2    4.4   
275      1992 Volkswagen Fox   2530.0      4.3         4.4      4.4    4.6   
276      1992 Volkswagen GTI   2789.0      4.5         4.7      4.6    4.7   
277  1992 Volkswagen Corrado   4507.0      4.8         4.9      4.9    4.8   

     Reliability  Styling integrated_garage_door_opener  max_seating_capacity  \
0            4.5      4.6            Port Not Installed                   7.0   
2            4.0      4.5            Port Not Installed                   5.0   
3            4.5      4.6            Port Not Installed                   7.0   
4            4.8      4.8            Port Not Installed                   5.0   
6            4.0      4.5            Port Not Installed                   5.0   
..           ...      ...                           ...                   ...   
273          4.6      4.4            Port Not Installed                   5.0   
274          4.1      4.2            Port Not Installed                   5.0   
275          4.7      4.2            Port Not Installed                   4.0   
276          4.8      4.6            Port Not Installed                   5.0   
277          4.7      4.9            Port Not Installed                   4.0   

     ... Drivetrain Engine max_years_warrantly max_miles_warrantly  \
0    ...        FWD     V6                   4             50000.0   
2    ...        FWD  4-Cyl                   4             50000.0   
3    ...        FWD     V6                   4             50000.0   
4    ...        FWD  4-Cyl                   4             50000.0   
6    ...        FWD  4-Cyl                   4             50000.0   
..   ...        ...    ...                 ...                 ...   
273  ...        FWD  4-Cyl                   2             24000.0   
274  ...        FWD  4-Cyl                   2             24000.0   
275  ...        FWD  4-Cyl                   2             24000.0   
276  ...        FWD  4-Cyl                   2             24000.0   
277  ...        FWD  4-Cyl                   2             24000.0   

    max_years_powertrain  max_miles_powertrain  trunk_cap_categ  Year  \
0                    6.0               70000.0    Very Spacious  2024   
2                    6.0               70000.0         Spacious  2023   
3                    6.0               70000.0    Very Spacious  2023   
4                    6.0               70000.0         Moderate  2023   
6                    6.0               70000.0         Spacious  2022   
..                   ...                   ...              ...   ...   
273                  5.0               50000.0         Moderate  1992   
274                  5.0               50000.0         Moderate  1992   
275                  5.0               50000.0            Small  1992   
276                  5.0               50000.0         Moderate  1992   
277                  5.0               50000.0         Moderate  1992   

          Brand    Model  
0         Acura      MDX  
2         Acura      RDX  
3         Acura      MDX  
4         Acura      TLX  
6         Acura      RDX  
..          ...      ...  
273  Volkswagen     Golf  
274  Volkswagen   Passat  
275  Volkswagen      Fox  
276  Volkswagen      GTI  
277  Volkswagen  Corrado  

[9713 rows x 23 columns]

In [50]:
data_cleaned.to_excel("data_cars.xlsx",
             sheet_name='Sheet_name_1') 